src/environments/array.js

```diff
@@ -16,39 +16,15 @@ import type Parser from "../Parser";
 import type {ParseNode, AnyParseNode} from "../parseNode";
 import type {StyleStr} from "../types";
 import type {HtmlBuilder, MathMLBuilder} from "../defineFunction";
-import type {Measurement} from "../units";
 
 // Data stored in the ParseNode associated with the environment.
-type AlignSpec = { type: "separator", separator: string } | {
+export type AlignSpec = { type: "separator", separator: string } | {
     type: "align",
     align: string,
     pregap?: number,
     postgap?: number,
 };
 
-export type ArrayEnvNodeData = {|
```

Comment:
So flattening the "array" `ParseNode` made all this disappear?  Cool!

Reply:
Not really. It's still in the second parameter of `parseArray()` (which is now destructured). It's the only place it was used.